src/templates/Practice4.tsx

```diff
@@ -0,0 +1,132 @@
+import { useForm} from 'react-hook-form'
+import { object, string,InferType } from 'yup'
+import { yupResolver } from '@hookform/resolvers/yup'
+import { useState} from 'react';
+import Button from '../components/Button'
+
+  const validationSchema = object().shape({
+    postalCode: string()
+      .required('必須項目です')
+      .matches(/^\d{7}$/, '7桁の数字を入力してください'),
+    prefecture: string().required('必須項目です'),
+    city: string()
+      .required('必須項目です'),
+  }).required();
```

Comment:
All these error message strings should be in a separate file and import to use them here. Now in case we need to adjust the message, it can be easily managed from one single place.

```tsx
// src/messages.ts
 export const ERROR_MESSAGES = {
    required: '必須項目です',
    not7digits: '7桁の数字を入力してください',
  }

// src/templates/Practice4.tsx
  const validationSchema = object().shape({
    postalCode: string()
      .required(ERROR_MESSAGE.required)
      .matches(/^\d{7}$/, ERROR_MESSAGE.not7digits),
    prefecture: string().required(ERROR_MESSAGE.required),
    city: string()
      .required(ERROR_MESSAGE.required),
  }).required();
```

Reply:
Made the changes